client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetIndexResponseTests.java

```diff
@@ -0,0 +1,176 @@
+            for (String index : serverSideResponse.getIndices()) {
+                MappingMetaData indexMapping = serverSideResponse.getMappings().get(index).get("_doc");
+                if (indexMapping != null) {
+                    assertEquals(indexMapping, parsedResponse.getMappings().get(index));
+                } else {
+                    assertEquals(new MappingMetaData(MapperService.SINGLE_MAPPING_NAME, Collections.emptyMap()),
+                            parsedResponse.getMappings().get(index));
+                }
+            }
+            assertEquals(serverSideResponse.getAliases(), parsedResponse.getAliases());
+            assertEquals(serverSideResponse.getSettings(), parsedResponse.getSettings());
+            assertEquals(serverSideResponse.defaultSettings(), parsedResponse.getDefaultSettings());
+
+        }
+
+    }
+
+    private static org.elasticsearch.action.admin.indices.get.GetIndexResponse createTestInstance() {
+        String[] indices = generateRandomStringArray(5, 5, false, false);
+        ImmutableOpenMap.Builder<String, ImmutableOpenMap<String, MappingMetaData>> mappings = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, List<AliasMetaData>> aliases = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, Settings> settings = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, Settings> defaultSettings = ImmutableOpenMap.builder();
+        IndexScopedSettings indexScopedSettings = IndexScopedSettings.DEFAULT_SCOPED_SETTINGS;
+        boolean includeDefaults = randomBoolean();
+        for (String index: indices) {
+            // rarely have no types
+            int typeCount = rarely() ? 0 : 1;
+            mappings.put(index, createMappingsForIndex(typeCount));
+
+            List<AliasMetaData> aliasMetaDataList = new ArrayList<>();
+            int aliasesNum = randomIntBetween(0, 3);
+            for (int i=0; i<aliasesNum; i++) {
+                aliasMetaDataList.add(GetAliasesResponseTests.createAliasMetaData());
+            }
+            CollectionUtil.timSort(aliasMetaDataList, Comparator.comparing(AliasMetaData::alias));
+            aliases.put(index, Collections.unmodifiableList(aliasMetaDataList));
+
+            Settings.Builder builder = Settings.builder();
+            builder.put(RandomCreateIndexGenerator.randomIndexSettings());
+            settings.put(index, builder.build());
+
+            if (includeDefaults) {
+                defaultSettings.put(index, indexScopedSettings.diff(settings.get(index), Settings.EMPTY));
+            }
+        }
+        return new org.elasticsearch.action.admin.indices.get.GetIndexResponse(
+            indices, mappings.build(), aliases.build(), settings.build(), defaultSettings.build()
+        );
+    }
+
+    public static ImmutableOpenMap<String, MappingMetaData> createMappingsForIndex(int typeCount) {
```

Comment:
To avoid duplicating this logic, would it be possible to pull out a test utilities class similar to `RandomCreateIndexGenerator` that lives in `test:framework`? This would also be useful to have available when working on 'get mappings'.